src/Orleans.Core/Async/AsyncLock.cs

```diff
@@ -54,9 +54,9 @@ public AsyncLock()
             semaphore = new SemaphoreSlim(1);
         }
 
-        public Task<IDisposable> LockAsync()
+        public Task<IDisposable> LockAsync(CancellationToken cancellation = default)
         {
-            Task wait = semaphore.WaitAsync();
+            Task wait = semaphore.WaitAsync(cancellation);
             if (wait.IsCompleted)
```

Comment:
IsCompleted is no longer sufficient here, since the task can now fail (by being canceled), and similarly, the below code to await completion via `ContinueWith` needs to be updated. Simplest update would be to return a call to a static async local function to await the task.